### PR TITLE
[libgit2] Update to 1.4.2

### PR DESCRIPTION
**Pull request by edo9300**
https://github.com/libgit2/libgit2/releases/tag/v1.4.1

### CODE REVIEW

**Comment by edo9300** [comment]
Changed this pr to a draft until this issue is figured out https://github.com/libgit2/libgit2/issues/6225 (it was introduced with 1.4.0). Up to that point it would only be harmful for the developers to push a broken library.

**Comment by edo9300** [comment]
The bugfix version 1.4.2 got released and fixes the issue https://github.com/libgit2/libgit2/releases/tag/v1.4.2, updated the files to build 1.4.2 now

File: ports/libgit2/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "libgit2",
-  "version-semver": "1.3.0",
-  "port-version": 2,
+  "version-semver": "1.4.2",
+  "port-version": 1,
```
Comment (JonLiu1993) [inline ports/libgit2/vcpkg.json:4]:
```suggestion
```

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@edo9300 ,Thanks for your pr, please run command "./vcpkg x-add-version libgit2 --overwrite-version" and commit again

**Review by JonLiu1993 — APPROVED** [review]

**Comment by edo9300** [comment]
> @edo9300 ,Thanks for your pr, please run command "./vcpkg x-add-version libgit2 --overwrite-version" and commit again

I missed the notification and I didn't notice your comment, sorry for that. Well since now you pushed it I guess I won't have to do anything.

**Pull request merged**